drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1270,6 +1272,35 @@ static void usbd_work_handler(struct k_work *item)
 			}
 			break;
 		case USBD_EVT_SOF:
+			if (nrfx_usbd_ep_enable_check(NRFX_USBD_EPOUT8)) {
+				/** Dissect receive request
+				 * if the iso OUT ep is enabled
+				 */
+				ep_ctx = endpoint_ctx(NRFX_USBD_EPOUT8);
+				if (!ep_ctx) {
+					LOG_ERR("There is not ISO ep");
+					break;
+				}
+				if (ep_ctx->cfg.en) {
+					ep_ctx->read_pending = true;
+					ep_ctx->read_complete = true;
```

Comment:
That looks weird to set both `read_pending` and `read_complete`

Reply:
TBH i do not remember why i did so, probably something was not working, i will check that, thanks.

Reply:
OK, I've checked that. I do not know why but when receive request is served (the one I am creating here) both of those flags are checked if they equal 'true' if not, the created event would not be served at all. I did not analyze why it was done like that but it (unluckily) must be here in order to make it work.